lib/std/special/compiler_rt/os_version_check_test.zig

```diff
@@ -0,0 +1,10 @@
+const __isPlatformVersionAtLeast = @import("os_version_check.zig").__isPlatformVersionAtLeast;
+const testing = @import("std").testing;
+
+test "isPlatformVersionAtLeast" {
```

Comment:
Move the test inside `os_version_check.zig` and have it execute only if the target OS is macos.

Reply:
Ah good call on only executing if target is macOS

normally I would've done test in same file, but I noticed compiler_rt had an unusual amount of `_test.zig` files and thought maybe some trickery was going on there.

Reply:
The reason for _test.zig files is that they are excluded from installation. Compiler-rt has a lot of MiB of tests.